components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -1224,6 +1227,23 @@ public int getRelativeDepthBetweenOrganizationsInSameBranch(String firstOrgId, S
         }
     }
 
+    @Override
+    public String getAnAncestorOrganizationIdInGivenDepth(String organizationId, int depth)
+            throws OrganizationManagementServerException {
+
+        try {
+            return Utils.getNewTemplate().fetchSingleRecord(
+                    GET_PARENT_ORGANIZATION_ID_WITH_DEPTH,
+                    (resultSet, rowNumber) -> resultSet.getString(1),
+                    namedPreparedStatement -> {
+                        namedPreparedStatement.setString(DB_SCHEMA_COLUMN_NAME_ID, organizationId);
+                        namedPreparedStatement.setInt(DB_SCHEMA_COLUMN_NAME_DEPTH, depth);
+                    });
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_CODE_GET_PARENT_IN_DEPTH, e, organizationId);
```

Comment:
Rename the constants with Ansestor since this is a generic function

Reply:
Updated